server/plugin/command.go

```diff
@@ -501,6 +518,163 @@ func (p *Plugin) handleIssue(_ *plugin.Context, args *model.CommandArgs, paramet
+					insecureVal, _ := strconv.ParseBool(val)
+					config["insecure_ssl"] = insecureVal
+				case strings.Contains(val, "application/json") || strings.Contains(val, "x-www-form-urlencoded"):
+					config["content_type"] = val
+				default:
+					if strings.Contains(val, "*") && len(val) == 1 {
+						hook.Events = []string{"*"}
+						break
+					}
+					if err := p.CheckOptionsValid(strings.Split(val, ",")); err != nil {
+						return err.Error()
+					}
+					hook.Events = strings.Split(val, ",")
+				}
+			}
+		}
+		if _, exists := config["url"]; !exists {
+			config["url"] = siteURL
+		}
+		if _, exists := config["insecure_ssl"]; !exists {
+			config["insecure_ssl"] = false
+		}
+		if _, exists := config["content_type"]; !exists {
+			config["content_type"] = "x-www-form-urlencoded"
+		}
+		if len(hook.Events) == 0 {
+			hook.Events = []string{"*"}
+		}
+		hook.Config = config
+		githubHook, _, err := p.CreateHook(ctx, githubClient, owner, repo, hook)
+		if err != nil {
+			return err.Error()
+		}
+
+		hookDetails, _, err := p.GetHook(ctx, githubClient, owner, repo, *githubHook.ID)
+		if err != nil {
+			return err.Error()
+		}
+
+		hookURL := baseURL + owner
+		if repo != "" {
+			hookURL += "/" + repo
+		}
+		hookURL += "/settings/hooks/"
+		txt := "Webhook Created Successfully \n"
+		hookID := strconv.Itoa(int(*githubHook.ID))
+		txt += fmt.Sprintf(" *  [%s](%s) :  -  %s \n", hookID, hookURL+hookID, strings.Join(hookDetails.Events, " , "))
+		return txt
+	case list:
+
+		if len(parameters) != 1 {
+			return "Unknown action, please use `/github help` to see all actions available."
+		}
+
+		owner, repo := parseOwnerAndRepo(parameters[0], p.getBaseURL())
+		var githubHooks []*github.Hook
+		var githubResponse *github.Response
+		var err error
+		opt := &github.ListOptions{
+			PerPage: 10,
+		}
+		if repo == "" {
+			for {
+				githubHooks, githubResponse, err = githubClient.Organizations.ListHooks(ctx, owner, opt)
+				if err != nil {
+					return err.Error()
+				}
+				if githubResponse.NextPage == 0 {
+					break
+				}
+				opt.Page = githubResponse.NextPage
+			}
+		} else {
+			for {
+				githubHooks, githubResponse, err = githubClient.Repositories.ListHooks(ctx, owner, repo, opt)
+				if err != nil {
+					return err.Error()
+				}
+				if githubResponse.NextPage == 0 {
+					break
+				}
+				opt.Page = githubResponse.NextPage
+			}
+		}
+
+		var txt string
+		if len(githubHooks) == 0 {
+			txt = "Currently there are no webhook in this [Owner/Repo]"
+		} else {
+			txt = "### Webhook in this Repositories\n"
+		}
+
+		for _, hook := range githubHooks {
+			hookID := strconv.Itoa(int(*hook.ID))
+			hookURL := baseURL + owner
+			if repo != "" {
+				hookURL += "/" + repo
+			}
+			hookURL += "/settings/hooks/"
+			txt += fmt.Sprintf(" *  [%s](%s) :  -  %s \n", hookID, hookURL+hookID, strings.Join(hook.Events, " , "))
+		}
+
+		return txt
+	}
+	return "Invalid action, only `add` and `list` commands are supported."
```

Comment:
Can we split `handlewebhook` into a few different functions, so this is a little more readable? Something like:

```go
switch command {
	case add:
            txt, err := p.handleWebhookAdd(args)
            // log err if not nil
            return txt
        case list:
            txt, err := p.handleWebhookList(args)
            // log err if not nil
            return txt
        default:
            return "Invalid action, only `add` and `list` commands are supported."
```

func (p *Plugin) handlewebhook(_ *plugin.Context, args *model.CommandArgs, parameters []string, userInfo *GitHubUserInfo) string {

Reply:
Sure , implemented this